standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java

```diff
@@ -254,8 +259,8 @@ private void startLocalSegmentStore(int segmentStoreId) throws Exception {
                         .with(ServiceConfig.LISTENING_PORT, this.segmentStorePorts[segmentStoreId])
                         .with(ServiceConfig.CLUSTER_NAME, this.clusterName)
                         .with(ServiceConfig.ENABLE_TLS, this.enableTls)
-                        .with(ServiceConfig.KEY_FILE, this.keyFile)
-                        .with(ServiceConfig.CERT_FILE, this.certFile)
+                        .withDefaultIfNotExists(ServiceConfig.KEY_FILE, this.keyFile)
```

Comment:
This is more along the lines of what I was after, but it does not look very good. Here are the options I see:

1- I'm not sure why the defaults for `this.keyFile` and `this.certFile` can't be empty string. If they are initialized with empty string and not set again, then we don't need to perform any of these gymnastics.
2- The other option is what @EronWright suggested, which is to make this more fluent. Right now there is a generic `with(...)` call, but we could specialized it so that we can have logic that target specific variables, e.g., `withKeyFile` and `withCertFile`.

Reply:
The problem with both those ideas is that they are special cases, either w.r.t. type or w.r.t. the specific property.  My earlier suggestion has neither of those problems:
```
public <V> ConfigBuilder<T> with(Property<V> property, Optional<V> maybeValue)
```

Reply:
I suppose I am saying that it would be nice to find a solution that allowed for fluency without relying on a sentinel value.

Reply:
@EronWright I think that might work better. @arvindkandhare what do you think?

Reply:
Yes. This makes sense. Updated the PR to take care of this.